homeassistant/components/shelly/entity.py

```diff
@@ -20,6 +20,30 @@ def temperature_unit(block_info: dict) -> str:
     return TEMP_CELSIUS
 
 
+def shelly_naming(self, block, entity_type: str):
+    """Naming for switch and sensors."""
+
+    if "num_outputs" not in self.wrapper.device.shelly.keys():
+        channels = 0
+    else:
+        channels = self.wrapper.device.shelly["num_outputs"]
+        if (
+            self.wrapper.model in ["SHSW-21", "SHSW-25"]
+            and self.wrapper.device.settings["mode"] == "roller"
+        ):
+            channels = 1
+    entity_name = self.wrapper.name
+    if channels > 1 and block.type != "device":
+        entity_name = self.wrapper.device.settings["relays"][int(block.channel)]["name"]
+        if not entity_name:
+            entity_name = f"{self.wrapper.name} channel {str(int(block.channel)+1)}"
+    if entity_type == "switch":
+        return entity_name
+    if entity_type == "sensor":
+        return f"{entity_name} {self.description.name}"
+    raise ValueError
```

Comment:
I miss the larger context in this, but this can be made more readable.
Also, `keys()` when checking for `num_outputs`, also casting to a string in a f-string is not needed.

```suggestion
    channels = 0
    if "num_outputs" in self.wrapper.device.shelly:
        channels = self.wrapper.device.shelly["num_outputs"]
        if (
            self.wrapper.model in ["SHSW-21", "SHSW-25"]
            and self.wrapper.device.settings["mode"] == "roller"
        ):
            channels = 1

    entity_name = self.wrapper.name
    if channels > 1 and block.type != "device":
        entity_name = self.wrapper.device.settings["relays"][int(block.channel)]["name"]
        if not entity_name:
            entity_name = f"{self.wrapper.name} channel {int(block.channel)+1}"

    if entity_type == "switch":
        return entity_name

    if entity_type == "sensor":
        return f"{entity_name} {self.description.name}"

    raise ValueError
```

Reply:
Thx a lot for the review